### ISSUE TITLE
Why does the FPS of a raw video go crazy?

### ISSUE THREAD

**Issue by Rangtian** [Franiac/TwitchLeecher#657]
Hi Franiac,

I've tried several extensions and websites, from which I downloaded twitch videos. However, the FPS rushes to 1000 when I play them locally!!!

The only one that works is Twitch Leecher. And I still have to convert the downloaded videos to ".mp4" through Twitch Leecher. Otherwise, the FPS will still be 1000 if I choose to download raw videos, which are ".ts" files.

Thank you.

![12345678](https://user-images.githubusercontent.com/22640407/144296750-a6434017-7bf6-4798-867d-b7cb3cd47320.png)

**Comment by Franiac** [comment]
1000fps where? Please provide more information, because I cannot follow what you are doing and where you measure 1000fps.

**Comment by Rangtian** [comment]
Here is the Download Log.

Download task has been started!

TWITCH LEECHER INFO
--------------------------------------------------------------------------------------------
Version: 1.8.4

VOD INFO
--------------------------------------------------------------------------------------------
VOD ID: 1218662480
Selected Quality: 1920x1080@60fps (Source)
Download Url: https://www.twitch.tv/videos/1218662480
Crop Start: No
Crop End: No

OUTPUT INFO
--------------------------------------------------------------------------------------------
Disable Conversion: Yes
Output File: C:\Users\rangtian\Downloads\20211129_1218662480_World of Tanks.ts
FFMPEG Path: C:\Program Files\Twitch Leecher\ffmpeg.exe
Temporary Download Folder: C:\Users\rangtian\AppData\Local\Temp\TL_4f765a8d-e41b-4ec6-a047-132059e46ebb

ACCESS INFO
--------------------------------------------------------------------------------------------
Token: %7B%22authorization%22%3A%7B%22forbidden%22%3Afalse%2C%22reason%22%3A%22%22%7D%2C%22chansub%22%3A%7B%22restricted_bitrates%22%3A%5B%5D%7D%2C%22device_id%22%3Anull%2C%22expires%22%3A1638490518%2C%22https_required%22%3Atrue%2C%22privileged%22%3Afalse%2C%22user_id%22%3Anull%2C%22version%22%3A2%2C%22vod_id%22%3A1218662480%7D
Signature: f468066afa82710c3a7ccc313109ad4d496e45e6
Sub-Only: No
Privileged: No

Creating temporary download directory 'C:\Users\rangtian\AppData\Local\Temp\TL_4f765a8d-e41b-4ec6-a047-132059e46ebb'... done!

Retrieving m3u8 playlist urls for all VOD qualities... done!
https://d1ymi26ma8va5x.cloudfront.net/63f96c87666f8ae0edcd_iyouxin_68267362434_6217325126/chunked/highlight-1218662480.m3u8
https://d1ymi26ma8va5x.cloudfront.net/63f96c87666f8ae0edcd_iyouxin_68267362434_6217325126/720p60/highlight-1218662480.m3u8
https://d1ymi26ma8va5x.cloudfront.net/63f96c87666f8ae0edcd_iyouxin_68267362434_6217325126/720p30/highlight-1218662480.m3u8
https://d1ymi26ma8va5x.cloudfront.net/63f96c87666f8ae0edcd_iyouxin_68267362434_6217325126/480p30/highlight-1218662480.m3u8
https://d1ymi26ma8va5x.cloudfront.net/63f96c87666f8ae0edcd_iyouxin_68267362434_6217325126/audio_only/highlight-1218662480.m3u8
https://d1ymi26ma8va5x.cloudfront.net/63f96c87666f8ae0edcd_iyouxin_68267362434_6217325126/360p30/highlight-1218662480.m3u8
https://d1ymi26ma8va5x.cloudfront.net/63f96c87666f8ae0edcd_iyouxin_68267362434_6217325126/160p30/highlight-1218662480.m3u8

Playlist url for selected quality 1920x1080@60fps (Source) is https://d1ymi26ma8va5x.cloudfront.net/63f96c87666f8ae0edcd_iyouxin_68267362434_6217325126/chunked/highlight-1218662480.m3u8

Retrieving playlist... done!
Parsing playlist... done!
Number of video chunks: 8

Starting parallel video chunk download
Number of video chunks to download: 8
Maximum connection count: 10

Parallel video chunk download is running...

Download of all video chunks complete!

Merging all VOD parts into 'C:\Users\rangtian\Downloads\20211129_1218662480_World of Tanks.ts'...

Starting temporary download folder cleanup!
Deleting directory 'C:\Users\rangtian\AppData\Local\Temp\TL_4f765a8d-e41b-4ec6-a047-132059e46ebb'... done!

Download task ended successfully!

**Comment by Franiac** [comment]
> From 00:00:11 the video rushes to 1000 FPS while the audio plays normally

Where does it rush to 1000fps? In VLC? While transcoding? While watching online? Please explain what you are doing, mind reading is not my strength...

**Comment by Rangtian** [comment]
> Where does it rush to 1000fps? In VLC? While transcoding? While watching online? Please explain what you are doing, mind reading is not my strength...

It happens while watching locally with PotPlayer, or any other media player that comes with Windows 10, after downloaded into my hard drive.

**Comment by Franiac** [comment]
The .ts file from Twitch is already corrupt. VLC manages to continue normal playback but it clearly hangs and skips at 11s.

Since I am an ex WoT pro player and know Iyouxin, I can tell you that his stream settings are pretty nasty and his internet connection drops out sometimes. That is nothing Twitch or I can fix.

**Comment by Rangtian** [comment]
It's not iyouxin's issue. I've tried to download skill4ltu's, still have the same issue. Have you tried to download without conversion to mp4? Don't you encounter the same issue?

![image](https://user-images.githubusercontent.com/22640407/144481749-1c390d85-5141-4504-b02a-914693942420.png)

**Comment by Franiac** [comment]
Yes, I disabled the conversion and that means the source file is corrupted. Iyouxins fault or not, I cannot fix broken streams, broker files or encoding errors. If TL receives broken stuff from Twitch, it stays broken, that's how it works.

**Comment by Rangtian** [comment]
> Yes, I disabled the conversion and that means the source file is corrupted. Iyouxins fault or not, I cannot fix broken streams, broken files or encoding errors. If TL receives broken stuff from Twitch, it stays broken, that's how it works.

I see. Thank you for your time.

I think it should be twitch's issue. Because my IDM has been working well for years, and suddenly every tool, extension, and website for downloading twitch has this 1000fps issue. Except yours.
